4.0 - Update changes.md file for the new API (#5042)
wazuh/wazuh: api/api/spec/changes.md

## General
* Date type use a standard format ISO-8601 defined by date-time format.
* Deleted all return parameters **path**, new API don't show any absolute path in responses.
* Changed search negation from `!` to `-`.
* Changed nested fields from `a_b` to `a.b`
* Changed parameter **query** to allow reserved characters.
* The endpoint's responses has been changed accordingly to the new RBAC standard. See spec schema responses carefully.
* The responses no longer will have `items` and `totalitems` fields, instead most responses will have the following structure:
```
{
  "data": {
    "affected_items": [],
    "total_affected_items": 0,
    "total_failed_items": 0,
    "failed_items": [],
    "message": ""
  }
}
```
* In responses, `msg` key is now moved to new `message` key

> * In responses, `filter` key is now moved to new `filters` key

Reviewer: Remove this line, I haven`t found any response that uses this key anymore.

Author: Done

* Errors follow the generic error format and are shown in `dapi_errors` key

## Default
### GET     /
* New endpoint. Returns basic information about the API.

## Active Response
### PUT     /active-response
* New endpoint. Run commands in all agents by default. 
* Use **list_agents** parameter in query to specify which agents must run the command.

### PUT     /active-response/{agent_id}
* Endpoint removed. Use `PUT /active-response?list_agents=agent_id` instead.

## Agents
### DELETE  /agents
* Removed **ids** query parameter. Use **list_agents** instead.
* Added **list_agents** parameter in query used to specify which agents must be deleted. 
* If no **list_agents** is provided in query all agents will be removed.

### DELETE  /agents/{agent_id}
* Endpoint removed. Use `DELETE /agents?list_agents=agent_id` instead

### DELETE  /agents/{agent_id}/group

> * Added **list_groups** parameter in query to specify an array of group's ID to remove from the agent.

Reviewer: ```suggestion
* Added **list_groups** parameter in query to specify an array of group IDs to remove from the agent.
```

* Removes the agent from all groups by default or a list of them if **list_groups** parameter is found.	

### DELETE  /agents/group
* New endpoint. Remove all agents assignment or a list of them from the specified group.
* Use `group_id` parameter in query to specify the group.

### DELETE  /agents/group/{group_id}
* Endpoint removed. Use `DELETE /agents/group?group_id=id` instead.

### DELETE  /agents/groups
* Endpoint removed. Use `DELETE /groups` instead.

### DELETE  /agents/groups/{group_id}
* Endpoint removed. Use `DELETE /groups?list_groups=group_id` instead.

### GET     /agents
* Return information about all available agents or a list of them.

> * Added parameter **list_agents** in query used to specify a list of agents IDs (separated by comma) from which agents get the information.

Reviewer: ```suggestion
* Added parameter **list_agents** in query used to specify a list of agent IDs (separated by comma) from which agents get the information.
```

* Added parameter **registerIP** in query used to filter by the IP used when registering the agent.

> * With this new endpoint, you won't get a 400 response in agent name cannot be found,

Reviewer: ```suggestion
* With this new endpoint, you won't get a 400 response if agent name cannot be found,
```
